File: localmart/order-service/src/main/java/com/localmart/order_service/event/PaymentProcessedEvent.java
package com.localmart.order_service.event;

import java.util.UUID;

/**
 * Event consumed from Kafka topic "payment.processed".
 *
 * Published by: payment-service after processing a payment.
 *
 * status: "SUCCESS" → order moves to CONFIRMED
 *         "FAILED"  → order moves to CANCELLED (Saga compensation)
 *
 * reason: null on success, human-readable failure reason on failure
 * (e.g. "Insufficient funds", "Card declined")
 */
public record PaymentProcessedEvent(
        UUID orderId,
        String status,
        String reason
) {}

File: localmart/payment-service/DTOs/PaymentProcessedEvent.cs
namespace payment_service.DTOs;

// Published to "payment.processed" topic after processing payment.
// Consumed by order-service's PaymentEventConsumer — field names must match exactly.
//
// status: "SUCCESS" or "FAILED" — matches order-service's string comparison.
// reason: null on success, failure reason on failure.
public record PaymentProcessedEvent(
    Guid OrderId,
    string Status,
    string? Reason
);
